vaadin-text-field-flow-demo/src/main/java/com/vaadin/flow/component/textfield/demo/TextFieldView.java

```diff
@@ -164,6 +165,31 @@ private void textFieldFocusShortcut() {
         this.addCard("Text field", "Focus shortcut usage", textField);
     }
 
+    private void textFieldSetPattern(){
+        // begin-source-example
+        // source-example-heading: Pattern usage
+        TextField zipCode = new TextField("ZIP code");
+        zipCode.setPlaceholder("Only 3 letters codes are valid");
+        zipCode.setPattern("[A-Za-z]{3}");
+        zipCode.setPreventInvalidInput(true);
+
+        Paragraph status=new Paragraph();
+        TextField username = new TextField("Username");
+        username.setPattern("^[a-zA-Z0-9._-]{3,}");
+        username.addValueChangeListener(e->{
+            if(username.isInvalid())
+                status.setText("Username should be, at least, 3 character long " +
+                        "and contain only letters, digits, dashes or dots.");
+            else
+                status.setText("Your username seems valid!");
+        });
```

Comment:
Lots of formatting issues in this example, please use an autoformatter

Reply:
is there some maven goal for this? Or do you have rules somewhere?

Reply:
Here are the conventions https://github.com/vaadin/flow/tree/master/eclipse

Reply:
Done